### PR TITLE
ORLP-420 Add Javadocs to all interfaces (second part)

### PR DESCRIPTION
**Pull request by RomanSomyk**

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 1]
```diff
@@ -3,18 +3,51 @@
...
 
+/**
+ * This interface works with the comments to decks.
+ */
 public interface DeckCommentService {
-
+    /**
+     * Add a comment to decks
+     *
+     * @return true if operation is successful
```
Comment (voliynyk) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: `@return true` but method returns `DeckComment`

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 2]
```diff
@@ -3,18 +3,51 @@
...
+     */
     DeckComment addCommentForDeck(Long deckId, String commentText, Long parentCommentId)
             throws NotAuthorisedUserException;
 
+    /**
+     * Returns a comment with the given identifier
+     *
+     * @param commentId must not be {@literal null}.
```
Comment (voliynyk) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: as for me, if meaning of parameter is obvious, would be better to have `@param commentId comment id`, but try to find the same example in oracle code

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 3]
```diff
@@ -3,18 +3,51 @@
...
+     *
+     * @param deckId must not be {@literal null}.
+     * @return list of comments to the deck with the given identifier
+     */
     List<Comment> getAllCommentsForDeck(Long deckId);
 
+    /**
+     * Updates comment with the given identifier with the new text
+     *
+     * @param commentId   must not be {@literal null}
```
Comment (voliynyk) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: `@param commentId   must...` -> `@param commentId must...` just one space

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 4]
```diff
@@ -3,18 +3,51 @@
...
+     * @param commentText must not be {null}
+     * @return true if operation is successful
+     */
+
     DeckComment updateCommentById(Long commentId, String commentText);
 
+    /**
+     * Delete comments to the deck with the given identifier
```
Comment (voliynyk) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: `Delete comments to the deck with the given identifier` 
1. check grammar
2. `Delete comments` it means than this methods may delete few comments? or only one?

Comment (RomanSomyk) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: corrected

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java (hunk 3 above)
Comment (RomanSomyk) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: intelliji adding spaces while auto-format is using

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java (hunk 1 above)
Comment (RomanSomyk) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: corrected

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java (hunk 2 above)
Comment (RomanSomyk) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: corrected

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 5]
```diff
@@ -3,18 +3,56 @@
...
 
+/**
+ * This interface works with the comments to decks.
+ */
 public interface DeckCommentService {
-
+    /**
+     * Adds the comment to decks or answer to comment if @param parentCommentId is null.
+     *
+     * @param deckId          - decks id, must not be {@literal null}.
```
Comment (VovaChornyi) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: write 'deck's id' instead 'decks id'

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 6]
```diff
@@ -3,18 +3,56 @@
...
+
 import java.util.List;
 
+/**
+ * This interface works with the comments to decks.
+ */
 public interface DeckCommentService {
-
+    /**
+     * Adds the comment to decks or answer to comment if @param parentCommentId is not null.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: Adds the comment to 'deck'

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 7]
```diff
@@ -3,18 +3,56 @@
...
+    /**
+     * Gets the comment with the given identifier.
+     *
+     * @param commentId identifier of the comment, must not be {@literal null}.
+     * @return the comment with the given identifier.
+     */
     DeckComment getCommentById(Long commentId);
 
+    /**
+     * Gives all of the comments to the deck with the given identifier.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: Gets all comments of the deck with the given identifier.

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java [hunk 8]
```diff
@@ -4,9 +4,25 @@
 import com.softserve.academy.spaced.repetition.utils.exceptions.NotAuthorisedUserException;
 import com.softserve.academy.spaced.repetition.utils.exceptions.UserStatusException;
 
+/**
+ * This interface works with the decks rating.
+ * Only authorized users can rate the decks.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: Is it `line 9` necessary?

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java [hunk 9]
```diff
@@ -4,9 +4,25 @@
...
+ */
 public interface DeckRatingService {
-
+    /**
+     * Adds decks rating and checks if the user who rates deck is authorised and active.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: Adds deck rating

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java [hunk 10]
```diff
@@ -4,9 +4,25 @@
...
+     *
+     * @param rating value that user sets for deck, must not be more than 5 and less than 1.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: less than 1 and more than 5, maybe it is better, but it doesn't matter)

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java [hunk 11]
```diff
@@ -4,9 +4,25 @@
...
+     * @param deckId deck`s id for which rating is set, must not be {@literal null}.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: deck's id by which rating will set

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java [hunk 12]
```diff
@@ -3,18 +3,56 @@
...
+     *
+     * @param deckId deck`s id for which comments are searched, must not be {@literal null}.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckCommentService.java]: deck's id by which comments will searched

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 13]
```diff
@@ -5,46 +5,185 @@
 import com.softserve.academy.spaced.repetition.utils.exceptions.NotAuthorisedUserException;
 import com.softserve.academy.spaced.repetition.utils.exceptions.NotOwnerOperationException;
 import org.springframework.data.domain.Page;
+
 import java.util.List;
 
+/**
+ * This interface proceeds all operations with decks.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: maybe process insted of proceeds, but i am not sure in it

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 14]
```diff
@@ -5,46 +5,185 @@
...
+ */
 public interface DeckService {
+    /**
+     * Adds a new deck to the category
+     *
+     * @param deck       the deck which will be added to the category.
+     * @param categoryId category`s id for which the deck will be added.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: for which -> by which

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 15]
```diff
@@ -5,46 +5,185 @@
...
+     */
     void addDeckToCategory(Deck deck, Long categoryId);
 
+    /**
+     * Adds a new deck to the course
+     *
+     * @param deck       the deck which will be added to the category.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: to the course

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 16]
```diff
@@ -5,46 +5,185 @@
...
+     * @param categoryId id of the category for which belongs the course.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: for which -> to which, im not sure)

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 17]
```diff
@@ -5,46 +5,185 @@
...
+     * @param courseId   course`s id to which the deck will be added.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: to which -> by which

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 18]
```diff
@@ -5,46 +5,185 @@
...
+     */
     void addDeckToCourse(Deck deck, Long categoryId, Long courseId);
 
+    /**
+     * Updates the deck
+     *
+     * @param updatedDeck the new deck which will be set.
+     * @param deckId      id of the deck which will be updated.
+     * @param categoryId  category`s id for which the deck will be updated.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: set category id for updated deck ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 19]
```diff
@@ -5,46 +5,185 @@
...
+     *
+     * @param deckId id of the deck which will be deleted.
+     */
     void deleteDeck(Long deckId);
 
+    /**
+     * Creates new deck
+     *
+     * @param newDeck    new deck which will be added.
+     * @param categoryId category`s id for which the deck will be created.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: category id to which deck will be belonged ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java (hunk 17 above)
Comment (RomanSomyk) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: it`s the course id to which  we add the deck

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 20]
```diff
@@ -5,46 +5,185 @@
...
+     */
     void deleteOwnDeck(Long deckId)
             throws NotAuthorisedUserException, NotOwnerOperationException;
 
+    /**
+     * Updates the deck which was created by the user.
+     *
+     * @param updatedDeck the new deck which will be set.
+     * @param deckId      id of the deck which will be updated.
+     * @param categoryId  category`s id for which the deck will be updated.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: set category id of new deck

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 21]
```diff
@@ -5,46 +5,185 @@
...
+     * @throws NotAuthorisedUserException if unauthorized user creates new deck.
+     * @throws NotOwnerOperationException if the user is not owner of the deck updates the deck.
+     */
     Deck updateOwnDeck(Deck updatedDeck, Long deckId, Long categoryId)
             throws NotAuthorisedUserException, NotOwnerOperationException;
 
+    /**
+     * Gives all of the decks witch was created by the user.
+     *
+     * @return list of the decks witch was created by the user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: which

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 22]
```diff
@@ -5,46 +5,185 @@
...
+     * @throws NotAuthorisedUserException if unauthorized user gets the decks.
+     */
     List<Deck> getAllDecksByUser() throws NotAuthorisedUserException;
 
+    /**
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: Delete this JavaDoc

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 23]
```diff
@@ -5,46 +5,185 @@
...
+     *
+     *
+     *
+     */
     Deck getDeckUser(Long deckId) throws NotAuthorisedUserException, NotOwnerOperationException;
 
+    /**
+     * Gives the page with decks which are sorted by category.
+     *
+     * @param categoryId category id for which the decks will be given.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: for which -> by which

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 24]
```diff
@@ -5,46 +5,185 @@
...
+     * @param pageNumber the pages number on which there is the category.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: the page number on which will be displayed decks ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 25]
```diff
@@ -5,46 +5,185 @@
...
+     *
+     * @param deckId deck`s id for which syntax will highlight.
+     * @return highlighted syntax.
+     */
     String getSynthaxToHightlight(long deckId);
 
+    /**
+     * Gets the deck with the given identifier.
+     *
+     * @param deckId the deck which will searched.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: find deck by this id ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 26]
```diff
@@ -5,46 +5,185 @@
...
+     * @return the deck with the given identifier.
+     */
     Deck getDeck(Long deckId);
 
+    /**
+     * Gets the list of the decks which belongs to course with given identifier.
+     *
+     * @param courseId course`s id for which the decks will be searched.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: for which -> by which

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 27]
```diff
@@ -5,46 +5,185 @@
...
+     * @return the list of the decks which belongs to course with given identifier.
+     */
     List<Deck> getAllDecks(Long courseId);
 
+    /**
+     * Gets the list of the decks which belongs to category with given identifier.
+     *
+     * @param categoryId category`s id for which the decks will be searched.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: category id by which decks will be found

File: src/main/java/com/softserve/academy/spaced/repetition/service/FolderService.java [hunk 28]
```diff
@@ -2,9 +2,17 @@
...
 import com.softserve.academy.spaced.repetition.domain.Deck;
 import com.softserve.academy.spaced.repetition.utils.exceptions.NotAuthorisedUserException;
+
 import java.util.List;
 
 public interface FolderService {
+    /**
+     * Adds a new deck.
+     *
+     * @param deckId deck
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/FolderService.java]: id of the deck

File: src/main/java/com/softserve/academy/spaced/repetition/service/RegistrationService.java [hunk 29]
```diff
@@ -5,7 +5,17 @@
 import org.springframework.mail.MailException;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
+/**
+ * This interface supports the registration of the new user
+ */
 public interface RegistrationService {
+    /**
+     * Registers new users
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/RegistrationService.java]: user or users?

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 30]
```diff
@@ -34,49 +37,193 @@
 
     void addUser(User user);
 
+    /**
+     * Finds users by email.
+     *
+     * @param email user`s email which will be searched.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: email by which user will be found

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 31]
```diff
@@ -34,49 +37,193 @@
...
+     * @return user that was found.
+     * @deprecated newer used in project.
+     */
     User findUserByEmail(String email);
 
+    /**
+     * Gets all users from entity User.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: maybe without `from entity User`

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 32]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @return list of users.
+     * @deprecated newer used in project.
+     */
     List<User> getAllUsers();
 
+    /**
+     * Sets user-status as active.
+     *
+     * @param id user`s id which will be set as active.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: user id by which user status will be set as active

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 33]
```diff
@@ -34,49 +37,193 @@
...
+     * @return user tat was set as active.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: tat -> that

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 34]
```diff
@@ -34,49 +37,193 @@
...
+     */
     User setUsersStatusActive(Long id);
 
+    /**
+     * Sets user-status as deleted.
+     *
+     * @param id user`s id which will be set as deleted.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: user id by which user status will be set as deleted

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 35]
```diff
@@ -34,49 +37,193 @@
...
+    /**
+     * Sets user-status as blocked.
+     *
+     * @param id user`s id which will be set as blocked.
+     * @return user tat was set as blocked.
+     */
     User setUsersStatusBlocked(Long id);
 
+    /**
+     * Finds users by identifier.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: users -> user

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 36]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @param userId user`s id which will be searched.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: user id by which user will be found

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 37]
```diff
@@ -34,49 +37,193 @@
...
+     * @return user that was found.
+     */
     User getUserById(Long userId);
 
+    /**
+     * Ads deck to users folder
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: Ads -> Adds

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 38]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @param userId user`s id
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: user id, without `'s` ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 39]
```diff
@@ -34,49 +37,193 @@
...
+     * @param deckId deck`s id
+     * @return managed by admin user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: user that was found by userId

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 40]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     *
+     *
+     *
+     */
+
     String getNoAuthenticatedUserEmail() throws NotAuthorisedUserException;
 
+    /**
+     * Gets user from authorised user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: Gets authorised user.

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 41]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @return user from entity.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: authorised user

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 42]
```diff
@@ -34,49 +37,193 @@
...
+     * @throws NotAuthorisedUserException if user is unauthorised.
+     */
     User getAuthorizedUser() throws NotAuthorisedUserException;
 
+    /**
+     * Gets all courses which was passed by identified user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: by identifier user -> by user

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 43]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @param userId user`s id.
+     * @return Set of courses which was passed by identified user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: set of courses on which user is subscribed

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 44]
```diff
@@ -34,49 +37,193 @@
...
+     */
     Set<Course> getAllCoursesByUserId(Long userId);
 
+    /**
+     * Deletes deck from users folder.
+     *
+     * @param userId users id.
+     * @param deckId decks id.
+     * @return managed by admin user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: return user

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 45]
```diff
@@ -34,49 +37,193 @@
...
+     */
     User removeDeckFromUsersFolder(Long userId, Long deckId);
 
+    /**
+     * Gets list of decks from the folder of the defined user
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: Gets list of decks from user folder

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 46]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @param userId user`s id.
+     * @return list managed by admin users.
+     */
     List<Deck> getAllDecksFromUsersFolder(Long userId);
 
+    /**
+     * Gets page with all users.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: Gets page with users.

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 47]
```diff
@@ -34,49 +37,193 @@
...
+     * @param ascending  the value that determines how the elements must be sorted on the page.
+     * @return page with all users.
+     */
     Page<User> getUsersByPage(int pageNumber, String sortBy, boolean ascending);
 
+    /**
+     * Sets new personal data.
+     *
+     * @param person new personal data that will be set.
+     * @return managed by admin user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: What does `managed by admin user` mean?

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 48]
```diff
@@ -34,49 +37,193 @@
...
+     * @throws ImageRepositorySizeQuotaExceededException image size is to large.
+     * @throws NotAuthorisedUserException                unauthorized user sets image.
+     */
+
     User uploadImage(MultipartFile file) throws ImageRepositorySizeQuotaExceededException,
             NotAuthorisedUserException;
 
+    /**
+     * Gets byte array of profile image.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: of -> from ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 49]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @return byte array of profile image.
+     * @throws NotAuthorisedUserException unauthorized user gets profile image.
+     */
     byte[] getDecodedImageContent() throws NotAuthorisedUserException;
 
+    /**
+     * Activates new accounts.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: account

File: src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java [hunk 50]
```diff
@@ -34,49 +37,193 @@
...
+     *
+     * @throws NotAuthorisedUserException unauthorized user used this operation.
+     */
     void activateAccount() throws NotAuthorisedUserException;
 
+    /**
+     * Deletes user`s account.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java]: without `'s` ( not sure )

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 51]
```diff
@@ -5,46 +5,179 @@
...
+     * @param categoryId  category`s id for which the deck will be updated.
+     * @return updated deck.
+     * @throws NotAuthorisedUserException if unauthorized user creates new deck.
+     * @throws NotOwnerOperationException if the user is not owner of the deck updates the deck.
+     */
     Deck updateOwnDeck(Deck updatedDeck, Long deckId, Long categoryId)
             throws NotAuthorisedUserException, NotOwnerOperationException;
 
+    /**
+     * Gives all of the decks witch was created by the user.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: which

**Pull request merged**